### ISSUE TITLE
bors-ng silently discards all batches on GitHub 422 errors

### ISSUE THREAD

**Issue by asbach** [bors-ng/bors-ng#1152]
We have observed the following bors-ng behavior running on our own bors instance:

Crash log in bors:
```
{{:badmatch,
  {:error, :push, 422,
   "{\"message\":\"Required status check \\\"continuous-integration/jenkins/pr-merge\\\" is expected. At least 1 approving review is required by reviewers with write access.\",\"documentation_url\":\"https://docs.github.com/enterprise/2.22/user/articles/about-protected-branches\"}"}},
 [
   {BorsNG.Worker.Batcher, :complete_batch, 3,
    [file: 'lib/worker/batcher.ex', line: 712]},
   {BorsNG.Worker.Batcher, :maybe_complete_batch, 1,
    [file: 'lib/worker/batcher.ex', line: 658]},
   {BorsNG.Worker.Batcher, :handle_cast, 2,
    [file: 'lib/worker/batcher.ex', line: 94]},
   {:gen_server, :try_dispatch, 4,
    [file: 'gen_server.erl', line: 637]},
   {:gen_server, :handle_msg, 6,
    [file: 'gen_server.erl', line: 711]},
   {:proc_lib, :init_p_do_apply, 3,
    [file: 'proc_lib.erl', line: 249]}
 ]}
```

In this particular case, the target branch is has the required reviews set, but this particular branch shouldn't be used with bors anways, so that part is not what I'm worried about, I would say working as intended.

The issue that is actually a problem is that bors will silently drop all in-flight batches when it encounters this error. We have not yet taken a closer look into why this happens, but I have seen similar reports in other issues, e.g. #703. The other issues only mention that one batch and not all are dropped. 
Any ideas where we could start taking a closer look into the code?

**Comment by tommilligan** [comment]
Hey @asbach, I believe this can be now closed as fixed by #1273. Please take a look and verify if it covers your use case, or provide more details if not.

**Comment by asbach** [comment]
Hello @tommilligan,
I've deployed a version of bors that includes your fix and the issue appears to be fixed. Thanks a lot :-)
